### PR TITLE
BLD: use nonvendor versioneer

### PR DESCRIPTION
**Pull request by fangchenli**
This PR:
1. move versioneer config to pyproject.toml
2. use nonvendored version of versioneer

After this, we could enable PEP 517.

### CODE REVIEW

**Comment by WillAyd** [comment]
@lithomas1

File: pandas/_version.py [hunk 1]
```diff
@@ -1,20 +1,25 @@
-# pylint: disable=consider-using-f-string
```
Comment (lithomas1) [inline pandas/_version.py:1]: Were changes to this file made by versioneer? I'm surprised non vendored mode didn't get rid of it. 
(I guess I'll just nuke it after we drop setuptools)

**Comment by WillAyd** [comment]
Does meson-Python require versioneer to be vendored?

File: pandas/_version.py (hunk 1 above)
Comment (fangchenli) [reply pandas/_version.py:1]: Yes, this is from versioneer.

**Comment by lithomas1** [comment]
> Does meson-Python require versioneer to be vendored?

Seems to work fine.

**Review by lithomas1 — APPROVED** [review]
LGTM. Thanks for the cleanup.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Awesome, thanks @fangchenli. Always great to not vendor things if we don't have to

**Pull request merged**